### ISSUE TITLE
reverse dns lookups for vpn adresses

### ISSUE THREAD

**Issue by chefkoch-de42** [gravitl/netmaker#454]
Hi, 
I tried to set reverse dns lookup for vpn addresses  via 
`resolvectl domain nm-test "~test" "88.88.10.in-addr.arpa"`
where nm-test is the interface and ~test the netmaker network name and dns domain. But it gets overwritten every time netclients run.

Is there any chance to get reverse lookup for vpn ips integrated in netclient?

Regards
Peter

**Comment by afeiszli** [comment]
@chefkoch-de42 how did you implement the postup command. Did you insert it into the PostUp field in the UI, or just add it directly on the nodes?

**Comment by chefkoch-de42** [comment]
I added it in node config in WebUI.

**Comment by afeiszli** [comment]
Ok, to confirm, you're just trying to set a different dns server than Netmaker for the private dns right?

**Comment by afeiszli** [comment]
If so, you just need to turn off "dnsmode" on the nodes. You're correct, when dnsmode is on, it will run that command on all machines as the last step, regardless of what is in postup.

**Comment by chefkoch-de42** [comment]
No I just want to enhance the vpn address resolution with the reverse lockup. 
Netclient manipulates the clients resolved config via resolvectl (Iguess), but it adds just the forward lookup for the "Network domain". I'm trying to get the reverse lookup also. With the core DNS from netmaker.

**Comment by afeiszli** [comment]
Ok, that makes sense as a feature request then. In the short term, a workaround would be to turn off dns on each node by editing the netconfig file (dnson=no) and pushing (netclient push -n networkname). Then you can set both with resolvectl postup commands. Unfortunately we dont have a UI feature to set that field right now so it has to either be set locally during client creation (netclient join -t token -dnson no) or via the config file + pushing to server.

**Comment by chefkoch-de42** [comment]
thx for the workaround, but in fact it does not work as expected.
netclient config: `postup: iptables -A FORWARD -i nm-test -j ACCEPT; iptables -t nat -A POSTROUTING -o nm-test -j MASQUERADE; resolvectl domain nm-test "~test" "88.88.10.in-addr.arpa"`

results in 
`          DNS Domain: \034\126test\034
                      \03488.88.10.in-addr.arpa\034`

in resolvctl status. Netlient might mess up with ~ and "

**Comment by chefkoch-de42** [comment]
> Ok, that makes sense as a feature request then. 

That would be great

**Comment by afeiszli** [comment]
try with single quotes. Escape characters might work too (\").

**Comment by chefkoch-de42** [comment]
hmm, no same same but different:
`          DNS Domain: \039\126test\039
                      \03988.88.10.in-addr.arpa\039`

I tried without quotes, that works, even with ~

Working command: `    postup: iptables -A FORWARD -i nm-test -j ACCEPT; iptables -t nat -A POSTROUTING -o nm-test -j MASQUERADE; resolvectl domain nm-test ~test 88.88.10.in-addr.arpa`

**Comment by afeiszli** [comment]
Did you run the push command after setting?

**Comment by chefkoch-de42** [comment]
yes I did. I updated the config via `sed -r 's/dnson: "yes"/dnson: "no"/g' -i /etc/netclient/config/netconfig-test; netclient push -n test` to be sure there is no update from server.
When I run `netclient pull -n test` the config isn't changing also. 
But if I change any field in WebUI for this node the dnson is set to yes by next netclient run

**Comment by afeiszli** [comment]
hmm ok, that must be a bug then, we'll have to fix that in the next release.

**Comment by afeiszli** [comment]
adding to triage
